### PR TITLE
Add default completion commit characters

### LINKED ISSUE #1491: Completion list incorrectly includes full signatures for members
## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (1.0.1)

Product Information:
 Version:            1.0.1
 Commit SHA-1 hash:  005db40cd1

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.12
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.1
```

VS Code version:
1.13.0-insider (1.13.0-insider) 3bb9b5314ef50bd8225aba0565cadc6ca19bc821 2017-05-18T06:18:32.269Z

C# Extension version:
1.10.0-beta3 dc1908d0e984a52536289f34fe4be5f80eb94f10
## Steps to reproduce
1. Enter "if", it will show "if" snippet item in autocomplete list.
2. Enter space as part of "if (xxx)" expression, it will show methods list, "if" snippets will disappear.
3. Enter "(" as part of "if (xxx)" expression.
## Expected  behavior
Autocomplete with closing pair of parenthesis ")", or nothing. Even I try to insert "IF"-snippet with tab-key - it will do it only on second attempt (first one is just closing autocomplete popup). Totally not usable when you write condition expression in fast manner.
## Actual behavior
Automatic insertion of first item at autocomplete list.

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #1491 and #1487

### CODE REVIEW

File: src/features/completionItemProvider.ts [hunk 1]
```diff
@@ -14,6 +14,12 @@ import {CompletionItemProvider, CompletionItem, CompletionItemKind, Cancellation
 
 export default class OmniSharpCompletionItemProvider extends AbstractSupport implements CompletionItemProvider {
 
+    // copied from Roslyn here: https://github.com/dotnet/roslyn/blob/6e8f6d600b6c4bc0b92bc3d782a9e0b07e1c9f8e/src/Features/Core/Portable/Completion/CompletionRules.cs#L166-L169
+    private static DefaultCommitCharacters = [
+        ' ', '{', '}', '[', ']', '(', ')', '.', ',', ':',
+        ';', '+', '-', '*', '/', '%', '&', '|', '^', '!',
+        '~', '=', '<', '>', '?', '@', '#', '\'', '\"', '\\'];
+
```
Comment (jrieken) [inline src/features/completionItemProvider.ts]: That is one long list 😄 Unsure if all commit characters make sense on all completion item kinds but the API is used as indented.

Comment (DustinCampbell) [inline src/features/completionItemProvider.ts]: Those are precisely the characters we commit on in Visual Studio. :smile:

Comment (DustinCampbell) [inline src/features/completionItemProvider.ts]: Note that there are cases where we tweak the list for various commit characters for some items, but this matches 99.9% of them.

**Comment by DustinCampbell** [comment]
@filipw: Do you want to take a look?

**Comment by DustinCampbell** [comment]
Thanks!

**Pull request merged**